app/metal-controller-manager/cmd/agent/main.go

```diff
@@ -118,25 +146,64 @@ func main() {
 		log.Fatal(err)
 	}
 
+	var endpoint *string
+	if endpoint = procfs.ProcCmdline().Get(constants.AgentEndpointArg).First(); endpoint == nil {
+		log.Fatal(fmt.Errorf("no endpoint found"))
+	}
+
+	log.Printf("Using %q as API endpoint", *endpoint)
+
 	log.Println("Reading SMBIOS")
 
 	s, err := smbios.New()
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	var endpoint *string
-	if endpoint = procfs.ProcCmdline().Get("sidero.endpoint").First(); endpoint == nil {
-		log.Fatal(fmt.Errorf("no endpoint found"))
-	}
+	var (
+		register *string
+		wipe     *string
+	)
 
-	log.Printf("Creating resource via %q", *endpoint)
+	register = procfs.ProcCmdline().Get(constants.AgentRegisterArg).First()
+	if register != nil {
+		if err = create(*endpoint, s); err != nil {
+			log.Fatal(err)
+		}
 
-	if err = create(*endpoint, s); err != nil {
-		log.Fatal(err)
+		log.Println("Registration complete")
 	}
 
-	log.Println("Discovery complete")
+	wipe = procfs.ProcCmdline().Get(constants.AgentResetArg).First()
+	if wipe != nil {
```

Comment:
For potentially-destructive things, I generally prefer an accept-list rather than a reject list.  (Is that the appropriate substitution for black-list, white-list?)